tests/unit_tests/test_time_series_any_time_series.py

```diff
@@ -0,0 +1,293 @@
+    # arrange
+    model, crs = example_model_and_crs
+    any_time_series = TimeSeries(model)
+    any_time_series.timestamps = timestamps
+    # act
+    result = any_time_series.step_duration(2)
+    # assert
+    assert result is None
+
+
+@pytest.mark.parametrize('timestamps', [['2014-01-01T00:00:00.588Z', '2014-01-03T00:00:00.588Z']])
+def test_step_duration_between_index(timestamps, example_model_and_crs):
+    # arrange
+    model, crs = example_model_and_crs
+    any_time_series = TimeSeries(model)
+    any_time_series.timestamps = timestamps
+    # act
+    result = any_time_series.step_duration(1)
+    # assert
+    assert result.duration.days == 2
+
+
+@pytest.mark.parametrize('timestamps', [['2014-01-01T00:00:00.588Z', '2014-01-03T00:00:00.588Z']])
+def test_step_days_none(timestamps, example_model_and_crs):
+    # arrange
+    model, crs = example_model_and_crs
+    any_time_series = TimeSeries(model)
+    any_time_series.timestamps = timestamps
+    # act
+    result = any_time_series.step_days(0)
+    # assert
+    assert result is None
+
+
+@pytest.mark.parametrize('timestamps', [['2014-01-01T00:00:00.588Z', '2014-01-03T00:00:00.588Z']])
+def test_step_days_number(timestamps, example_model_and_crs):
+    # arrange
+    model, crs = example_model_and_crs
+    any_time_series = TimeSeries(model)
+    any_time_series.timestamps = timestamps
+    # act
+    result = any_time_series.step_days(1)
+    # assert
+    assert result == 2
+
+
+@pytest.mark.parametrize('timestamps', [['2014-01-01T00:00:00.588Z', '2014-01-03T00:00:00.588Z']])
+def test_add_timestamp_before(timestamps, example_model_and_crs):
+    # arrange
+    model, crs = example_model_and_crs
+    any_time_series = TimeSeries(model)
+    any_time_series.timestamps = timestamps
+    new_timestamp = '2013-01-01T00:00:00.588Z'
+    # act
+    any_time_series.add_timestamp(new_timestamp, allow_insertion = True)
+    # assert
+    assert any_time_series.timestamps[0] == new_timestamp
+    assert any_time_series.timestamps[1] == '2014-01-01T00:00:00.588Z'
+    assert any_time_series.timestamps[2] == '2014-01-03T00:00:00.588Z'
+
+
+@pytest.mark.parametrize('timestamps', [['2014-01-01T00:00:00.588Z', '2014-01-03T00:00:00.588Z']])
+def test_add_timestamp_after(timestamps, example_model_and_crs):
+    # arrange
+    model, crs = example_model_and_crs
+    any_time_series = TimeSeries(model)
+    any_time_series.timestamps = timestamps
+    new_timestamp = '2015-01-01T00:00:00.588Z'
+    # act
+    any_time_series.add_timestamp(new_timestamp, allow_insertion = True)
+    # assert
+    assert any_time_series.timestamps[0] == '2014-01-01T00:00:00.588Z'
+    assert any_time_series.timestamps[1] == '2014-01-03T00:00:00.588Z'
+    assert any_time_series.timestamps[2] == new_timestamp
+
+
+@pytest.mark.parametrize('timestamps', [['2014-01-01T00:00:00.588Z', '2014-01-03T00:00:00.588Z']])
+def test_add_timestamp_between(timestamps, example_model_and_crs):
```

Comment:
Might also be good to test that an exception is raised if allow_insertion is False with this data.